src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -171,6 +171,34 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
     }
     
     /**
+     * Generates the feedback session closed emails for the given {@code session}.
+     * @throws EntityDoesNotExistException
+     */
+    public List<EmailWrapper> generateFeedbackSessionClosedEmails(FeedbackSessionAttributes session) {
+
+        boolean isEmailNeededForStudents = false;
+        try {
+            isEmailNeededForStudents = fsLogic.isFeedbackSessionHasQuestionForStudents(
+                    session.getFeedbackSessionName(), session.getCourseId());
+        } catch (EntityDoesNotExistException e) {
+            log.severe("Course " + session.getCourseId() + " does not exist or "
+                    + "session " + session.getFeedbackSessionName() + " does not exist");
+        }
```

Comment:
Move all these in the `if (!session.isPrivateSession()) {` block because this whole process is actually redundant if the session is private.